spec/lib/advanced_search_builder_spec.rb

```diff
@@ -57,4 +57,47 @@ def initialize(blacklight_config)
       end
     end
   end
+
+  describe "#add_advanced_search_to_solr" do
+    let(:to_solr) { { q: 'advanced query', fq: 'inclusive facet' } }
+    let(:advanced_query) { double('advanced_query') }
+    let(:solr_params) { { q: 'basic', fq: 'original fq' } }
+    before do
+      allow(advanced_query).to receive(:to_solr).and_return(to_solr)
```

Comment:
Yeah, give it a try. When I take the stubbing out, I'm not seeing the same results as with it.

The other thing I'm confused about is what the expected behavior is. I'd think we'd want to  use the user's new query to narrow the current results, not overwrite any of the non-facet fields?